server/utils/auth/command_gate.py

```
@@ -142,6 +142,67 @@ def gate_command(
+        version_key = f"tool_perms_version:{org_id}"
+        current_version = rc.get(version_key)
+        if not current_version:
+            return
+        cached_version = getattr(state, "_perms_version", None)
+        if cached_version == current_version:
+            return
+        from utils.db.connection_pool import db_pool
+        from utils.auth.stateless_auth import set_rls_context
+        user_id = getattr(state, "user_id", None)
+        with db_pool.get_connection() as conn:
+            with conn.cursor() as cur:
```

Ruff (0.15.12)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)